File: 2024/july/java/23.java
class Solution {
    public int[] frequencySort(int[] nums) {
        HashMap<Integer, Integer> f = new HashMap<>();
        for (int cur : nums){
            f.put(cur , f.getOrDefault(cur , 0)+1);
        }
        var id = new Integer[201];
        for (int i = 0, k = -100; i < 201; ++i , ++k){
            id[i] = k;
        }

        Arrays.sort(id , new Comparator<Integer>(){
            public int compare(Integer a , Integer b){
                int fA = f.getOrDefault(a , 0);
                int fB = f.getOrDefault(b , 0);
                if (fA == fB) return b -a;
                return fA - fB;
            }
        });
        int k = 0;
        int n = nums.length;
        var ans = new int[n];
        for (int a : id){
            int sz = f.getOrDefault(a , 0);
            for (int i = 0; i < sz; ++i){
                ans[k++] = a;
            }
        }
        return ans;
    }
}

File: 2024/july/javaScript/23.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var frequencySort = function(nums) {
        let f = new Map();
        for (const cur of nums){
            if (f.has(cur)) f.set(cur , f.get(cur)+1);
            else f.set(cur , 1);
        }
        const id = Array.from({length : 201} , (_ , i) => i - 100);
        for (let i = -100; i <= 100; ++i) if (!f.has(i)) f.set(i , 0);
       
 
        id.sort((a , b) => {
            let fA = f.get(a , 0);
            let fB = f.get(b , 0);
            if (fA == fB) return b -a;
            return fA - fB;
        });

      
        let n = nums.length;
        let ans = [];
        for (const a of  id){
            if (!f.has(a)) continue;
            let sz = f.get(a);
            for (let i = 0; i < sz; ++i){
               ans.push(a);
            }
        }
        return ans;
};
